### ISSUE TITLE
Regarding the successful reproduction of results on the lolv2-s dataset, but failure on the other two datasets:

### ISSUE THREAD

**Issue by siyawasei** [albrateanu/LYT-Net#39]
Dear Professor, I have been unable to achieve the GT-PSNR metrics reported in your README on the other two datasets. I would like to ask if there are any modifications or adjustments needed in the code. I noticed that the image preprocessing rotation part defined in the code does not seem to be used. I am training the model on a 3060 GPU, and after many attempts, the results on the lolv1 dataset remain around 26.2 GT-PSNR. I would greatly appreciate it if you could provide some guidance. Thank you very much!

**Comment by siyawasei** [comment]
I am using the TensorFlow architecture, so I also use TensorFlow's data for comparison.

**Comment by siyawasei** [comment]
```python
def call(self, inputs):
   
    inputs_normalized = (inputs + 1.0) / 2.0
    ycbcr = tf.image.rgb_to_yuv(inputs_normalized)
    y, cb, cr = tf.split(ycbcr, 3, axis=-1)

I found that after modifying this code, the PSNR value tested on the lolv1 dataset is higher than before.
I want to know if there is a problem with the code.

**Comment by siyawasei** [comment]
2. model/losses.py — VGG ImageNet Normalization
Modification Location
losses.py:18-27 (within the perceptual_loss function)

Before Modification
python
def perceptual_loss(y_true, y_pred, loss_model):
    return tf.reduce_mean(tf.square(loss_model(y_true) - loss_model(y_pred)))
After Modification
python
def perceptual_loss(y_true, y_pred, loss_model):
    # ImageNet normalization: statistics expected by VGG
    # Convert [0,1] images to VGG expected format (subtract ImageNet mean, divide by std)
    mean = tf.constant([123.68, 116.779, 103.53], dtype=tf.float32)
    std = tf.constant([58.395, 57.12, 57.375], dtype=tf.float32)

    y_true_vgg = (y_true - mean) / std
    y_pred_vgg = (y_pred - mean) / std

    return tf.reduce_mean(tf.square(loss_model(y_true_vgg) - loss_model(y_pred_vgg)))
Reason for Modification
VGG19 is pre-trained on ImageNet and expects input data to be normalized with specific ImageNet statistics. The original code directly fed [0,1] images into VGG without proper ImageNet normalization, resulting in inaccurate perceptual loss calculation.

3. scripts/find_gamma.py — Fixed max_val for Gamma Correction
Modification Location
find_gamma.py:4-10

Before Modification
python
def adjust_gamma(image, gamma=1.0):
    max_val = tf.reduce_max(image)
    image_normalized = image / max_val
    image_gamma_corrected = tf.pow(image_normalized, gamma)
    image_gamma_corrected = image_gamma_corrected * max_val
    return image_gamma_corrected
After Modification
python
def adjust_gamma(image, gamma=1.0):
    # Use fixed max_val=1.0 to ensure consistency of gamma correction
    max_val = 1.0
    image_normalized = image / max_val
    image_gamma_corrected = tf.pow(image_normalized, gamma)
    image_gamma_corrected = image_gamma_corrected * max_val
    return image_gamma_corrected
Reason for Modification
Using a fixed max_val=1.0 ensures consistency and reproducibility of gamma correction across different images. The original method that used per-image reduce_max could lead to inconsistent behavior

**Comment by albrateanu** [comment]
Hi @siyawasei. Indeed, the TensorFlow implementation shows inconsistencies, due to an earlier misunderstanding on GT-Mean computation.

Please use the PyTorch version. And I would suggest disregarding GT-Mean metrics of our method, and focusing only on the standard metrics reported. Those are more reliable.

**Comment by siyawasei** [comment]
Thank you very much for your reply! Your reply has been very helpful to me!

**Comment by siyawasei** [comment]
I just need to compare your metrics in this column, right?

<img width="925" height="217" alt="Image" src="https://github.com/user-attachments/assets/9223b889-cb3f-4c01-a856-a89641eb5c20" />

**Comment by siyawasei** [comment]
@albrateanu Sorry to bother you again.
I trained the LOLv1 dataset on the PyTorch version and got a standard PSNR of around 21.4. I would like to know what the standard metrics are for PyTorch. Thank you very much.

**Comment by siyawasei** [comment]
I used the weights you provided and got these values during testing. I can compare them with the values of your weights, right?
Raw-PSNR: 21.609808, GT-PSNR: 26.629027
Raw-SSIM: 0.812420, GT-SSIM: 0.834447

**Comment by albrateanu** [comment]
I believe that is appropriate. I would suggest retraining without GT-Mean evaluation. Should get you better metrics.

**Comment by siyawasei** [comment]
@albrateanu Thank you again for your reply.

**Comment by lidaxia223** [comment]
So what are the metrics for training with the source code? Why did my metrics improve only in the first 100 epochs, but stop improving after the last 1400 epochs? My PSNR on LOLv1 using PyTorch and TensorFlow frameworks is around 25.5, which seems to indicate that the training process doesn't significantly improve the pre-trained model.
